server/src/test/java/io/crate/expression/udf/UserDefinedFunctionsTest.java

```diff
@@ -87,6 +90,41 @@ public void testOverloadingBuiltinFunctions() {
             .isLiteral(DummyFunction.RESULT);
     }
 
+    @Test
+    public void test_function_name_and_schema_with_camel_case_strings() throws IOException {
+        registerUserDefinedFunction(
+            DUMMY_LANG.name(),
+            "MyScHEmA",
+            "mYFunCTioN",
+            DataTypes.INTEGER,
+            List.of(DataTypes.INTEGER, DataTypes.INTEGER),
+            "function mYFunCTioN(a, b) { return a + b; }");
+
+        assertThatThrownBy(() -> sqlExecutor.asSymbol("MyScHEmA.mYFunCTioN(1::integer, 2::integer)"))
+            .isExactlyInstanceOf(UnsupportedFunctionException.class)
+            .hasMessage("Unknown function: myschema.myfunction(1, 2)");
+
+        Function f = (Function) sqlExecutor.asSymbol("\"MyScHEmA\".\"mYFunCTioN\"(ints, 10)");
+        assertThat(f.toString(Style.QUALIFIED)).isEqualTo("\"MyScHEmA\".\"mYFunCTioN\"(doc.users.ints, 10)");
+        assertThat(f.toString(Style.UNQUALIFIED)).isEqualTo("\"MyScHEmA\".\"mYFunCTioN\"(ints, 10)");
+
+        registerUserDefinedFunction(
```

Comment:
Let's add a comment that second sub-test is for no-schema case?

Reply:
thx, added.